Clean up of code

MarcusTantakoun opened the pull request:
Cleaned up code (using black), updated doc website, added some test files (in /tests)

haz commented:
Big one! Looked through a bunch of different parts, and looks pretty good. Can't go through it all, though. This ready for merge, or are you still picking away at things? What was with the removal of the parse.py in tests?

MarcusTantakoun commented:
Sorry about all the big pushes, I wanted to clean the code up (naming, argument requirements, etc.) so it resulted in a few bugs in the paper reconstructions. But I fixed it now and updated the docs page to date. Should be good to go now.
